cmd/secret/cmd/root.go

```diff
@@ -151,26 +181,44 @@ func (r *root) getCmd() *cobra.Command {
 	getCmd.Flags().StringVarP(&key, "key", "k", key, "key for pair key-value")
 	getCmd.Flags().StringVarP(&cipherKey, "cipher-key", "c", cipherKey, "cipher key for data encryption and decryption")
 	getCmd.Flags().StringVarP(&path, "path", "p", "file.txt", "the place where the value will be got")
+	getCmd.Flags().StringVarP(&redisURL, "redis-url", "r", "", "redis url address. Example: localhost:6379")
 
 	return getCmd
 }
 
 func (r *root) serverCmd() *cobra.Command {
 	var path string
 	var port string
+	var redisURL string
 	var serverCmd = &cobra.Command{
 		Use:   "server",
 		Short: "Run server runner mode to start the app as a daemon",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			logger := r.logger.Named("server")
-			ds, err := storage.NewFileVault(path)
+			rdb := redis.NewClient(&redis.Options{Addr: redisURL, Password: "", DB: 0})
+			err := rdb.Ping(r.cmd.Context()).Err()
 			if err != nil {
-				return fmt.Errorf("can't get storage by path: %w", err)
+				return fmt.Errorf("redis db is not reachable:  %w", err)
 			}
 			store := make(map[string]api.MethodFactoryFunc)
-			store["local"] = func(cipher string) (secretApi.Provider, func()) {
-				cr := crypto.NewCryptographer([]byte(cipher))
-				return provider.NewProvider(cr, ds), nil
+
+			if redisURL != "" {
+				// remote method set handler for redis storage
+				store["remote"] = func(cipher string) (secretApi.Provider, func()) {
+					dataRedis := storage.NewRedisVault(rdb)
+					cr := crypto.NewCryptographer([]byte(cipher))
```

Comment:
No need to initialize storages when new cipher is added: you could keep it outside of the function.